aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+
+  /**
+   * constructor with default glue client
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, String catalogId, String warehousePath) {
+    this(conf, GlueClient.create(), catalogId, warehousePath);
+  }
+
+  /**
+   * constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, GlueClient glue, String catalogId, String warehousePath) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.warehousePath = warehousePath;
+    validateWarehousePath();
+  }
+
+  private void validateWarehousePath() {
+    Preconditions.checkArgument(warehousePath != null && warehousePath.length() > 0,
+        "Glue warehouse path must be set");
+    Preconditions.checkArgument(warehousePath.charAt(warehousePath.length() - 1) != '/',
+        "Glue warehouse path %s must not end with /", warehousePath);
+  }
+
+  @Override
+  public void close() throws IOException {
+    glue.close();
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    String dbName = IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace());
+    String tableName = tableIdentifier.name();
+    return new GlueTableOperations(conf, glue, catalogId, dbName, tableName);
+  }
+
+  /**
+   * Use the same format as Hive for default warehouse location
+   * @param tableIdentifier table id
+   * @return default warehouse path
+   */
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    // check if value is set in database
+    GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+        .name(IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()))
+        .build());
+    String dbLocationUri = response.database().locationUri();
+    if (dbLocationUri != null) {
+      return String.format("%s/%s", dbLocationUri, tableIdentifier.name());
+    }
+
+    return String.format(
+        "%s/%s.db/%s",
+        warehousePath,
+        IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace()),
+        tableIdentifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    namespaceExists(namespace);
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<TableIdentifier> results = new ArrayList<>();
+    while (true) {
```

Comment:
I tend to be afraid of while(true) break, I would prefer this was written in a different style. One quick change here which wouldn't be that different would be a "do while" which would let you keep basically everything the same but you could drop the "break".

Ideally I like to avoid "break" if at all possible and especially in loops, but ymmv.

Reply:
Sure, will use a do...while syntax for this.